app/uk/gov/hmrc/nisp/controllers/ExclusionController.scala

```diff
@@ -40,7 +41,13 @@ trait ExclusionController extends NispFrontendController with AuthorisedForNisp
   def showSP: Action[AnyContent] = AuthorisedByAny.async { implicit user => implicit request =>
     val nino = user.nino.getOrElse("")
     nispConnector.connectToGetSPResponse(nino).map {
-      case SPResponseModel(_, Some(spExclusions: ExclusionsModel)) => Ok(excluded_sp(nino, spExclusions))
+      case SPResponseModel(Some(spSummary: SPSummaryModel), Some(spExclusions: ExclusionsModel), niExclusionOption) => {
```

Comment:
All arguments are coming as option, but we are not treating them as same way like -> Some(*) or directly as niExclusions?